app/scripts/background.js

```diff
@@ -390,17 +388,13 @@ browser.runtime.onConnectExternal.addListener(async (...args) => {
   await isInitialized;
   // This is set in `setupController`, which is called as part of initialization
 
-  ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-mmi)
-  connectExternalExtension(...args);
-  ///: END:ONLY_INCLUDE_IF
-  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
   const port = args[0];
-  if (port.sender.tab?.id) {
-    connectExternalCaip(...args);
-  } else {
+  const isDappConnecting = port.sender.tab?.id;
+  if (!process.env.MULTICHAIN_API || !isDappConnecting) {
```

Comment:
I'm having trouble following the branching here, why do we need the additional condition?

Reply:
Technically other extensions can already connect to the MetaMask extension directly over externally_connectable, but that currently serves the 1193 API. I think we want to continue serving the 1193 API to other extensions, otherwise this would be a breaking change

Reply:
Ah! Yes, that makes a ton of sense!

Reply:
Makes sense! Updated in [ea57e9a](https://github.com/MetaMask/metamask-extension/pull/27782/commits/ea57e9a46507f8cc830c6b282006f78e9f86cd28).